aea/registries/base.py

```diff
@@ -265,19 +345,25 @@ def register(
 
         :param item_id: a pair (skill id, item name).
         :param item: the item to register.
-        :param is_dynamicall_added: whether or not the item is dynamicall added.
+        :param is_dynamically_added: whether or not the item is dynamically added.
         :return: None
         :raises: ValueError if an item is already registered with that item id.
         """
         skill_id = item_id[0]
         item_name = item_id[1]
-        if item_name in self._items.get(skill_id, {}).keys():
+        skill_items = self._items.fetch(skill_id)
+        if skill_items is not None and item_name in skill_items.keys():
             raise ValueError(
-                "Item already registered with skill id '{}' and name '{}'".format(
-                    skill_id, item_name
-                )
+                f"Item already registered with skill id '{skill_id}' and name '{item_name}'"
             )
-        self._items.setdefault(skill_id, {})[item_name] = item
+
+        if skill_items is not None:
+            self._items.unregister(skill_id)
```

Comment:
Why did you add this?

Reply:
because we update it later by registering it again, but only in case we didn't find a record.